project/src/main/webapp/gmail.js

```diff
@@ -0,0 +1,365 @@
+//   * Criteria 2: Isn't an empty email
+//   * Criteria 3: Isn't a google chat message
+//   */
+// function isActualEmail(emailBodyValue, emailDateValue) {
+//     if (emailBodyValue.indexOf("Daily Insider") != -1) {
+//         return false;
+//     }
+//     if (emailDateValue.length <= 1) {
+//         return false;
+//     }
+//     if (emailBodyValue.indexOf("format=flowed") != -1) {
+//         return false;
+//     }
+//     return true;
+// }
+
+// function getEmailObjects() {
+//     console.log(emailObjects);
+// }
+
+// /**
+//  * Get Message with given ID.
+//  *
+//  * @param  {String} messageId ID of Message to get.
+//  */
+// function getMessage(messageId) {
+//     const messageRequest = gapi.client.gmail.users.messages.get({
+//         'userId': 'me',
+//         'id': messageId,
+//         'format': "raw"
+//     });
+//     messageRequest.execute(response => {
+//         //Convert from base64 encoding to text.
+//         const decodedEmail = atob(response.raw.replace(/-/g, '+').replace(/_/g, '/'));
+//         const emailBodyValue = getEmailBody(decodedEmail);
+//         const emailDateValue = getEmailDate(decodedEmail);
+//         const messageID = getMessageID(decodedEmail);
+//         if (isActualEmail(emailBodyValue, emailDateValue)) {
+//             emailObjects[messageID] = { emailDate: emailDateValue, emailBody: emailBodyValue };
+//         }
+//         getEmailObjects();
+//     });
+// }
+
+// //Retrieve messages using hardcoded queries and the signed-in email.
+// function listMessages() {
+//     emailObjects = {};
+//     var getPageOfMessages = function(request, result) {
+//         request.execute(function(resp) {
+//             result = result.concat(resp.messages);
+//             var nextPageToken = resp.nextPageToken;
+//             if (nextPageToken) {
+//                 request = gapi.client.gmail.users.messages.list({
+//                     'userId': 'me',
+//                     'pageToken': nextPageToken,
+//                     'q': getQuery()
+//                 });
+//                 getPageOfMessages(request, result);
+//             } else {
+//                 for (var i = 0; i < result.length; i++) {
+//                     if (result[i] === undefined) {
+//                         console.log("There are no messages from the query");
+//                     }
+//                     else {
+//                         console.log(result.length);
+//                         console.log(result);
+//                         this.getMessage(result[i].id);
+//                     }
+//                 }
+//             }
+//         });
+//     };
+//     var initialRequest = gapi.client.gmail.users.messages.list({
+//         'userId': 'me',
+//         'q': getQuery()
+//     });
+//     getPageOfMessages(initialRequest, []);
+// }
+
+class gmailAPI {
+    question;
+    emailObjects = {};
+
+    constructor(question) {
```

Comment:
It's a little odd that the question is passed into the constructor here. It makes it seem like there's one gmailAPI instance per question, but that's not the case. I think question is only used by listMessages anyway, so can you just make it a parameter for listMessages?

Reply:
I made a gmail.getQuery instead that stores it into the gmail question variable and then listMessages doesn't need the question anymore.